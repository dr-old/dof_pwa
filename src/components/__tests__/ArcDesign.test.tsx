import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import { ThemeProvider, createTheme } from "@mui/material/styles";
import ArcDesign from "../ArchDesign";

describe("ArcDesign Component", () => {
  // Create a theme for testing inside the test function
  const theme = createTheme();

  test("renders correctly and matches snapshot", () => {
    const { asFragment } = render(
      <ThemeProvider theme={theme}>
        <ArcDesign backgroundColor="#fff" title="Performance" />
      </ThemeProvider>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  test("displays the correct title and details", () => {
    render(
      <ThemeProvider theme={theme}>
        <ArcDesign backgroundColor="#fff" title="Performance" />
      </ThemeProvider>
    );

    expect(screen.getByText("Performance")).toBeInTheDocument();
    expect(screen.getByText("More")).toBeInTheDocument();
    expect(screen.getByText("Social Media")).toBeInTheDocument();
    expect(screen.getByText("78%")).toBeInTheDocument();
    expect(screen.getByText("Organic Search")).toBeInTheDocument();
    expect(screen.getByText("22%")).toBeInTheDocument();
    // Add any additional checks if needed
  });
});
