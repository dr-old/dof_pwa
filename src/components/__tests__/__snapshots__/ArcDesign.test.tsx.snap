// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArcDesign Component renders correctly and matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-1exnj3w"
  >
    <div
      class="MuiBox-root css-1ialerq"
    >
      <h6
        class="MuiTypography-root MuiTypography-subtitle1 css-qjult-MuiTypography-root"
      >
        Performance
      </h6>
      <span
        class="MuiTypography-root MuiTypography-caption css-1muhxyk-MuiTypography-root"
      >
        More
      </span>
    </div>
    <div
      class="MuiBox-root css-3maudg"
    >
      <div
        aria-valuemax="100"
        aria-valuemin="0"
        aria-valuenow="60"
        class="MuiGauge-root css-vl6h9y-MuiGauge-container"
        role="meter"
      >
        <svg
          aria-hidden="true"
          class="css-1evyvmv-MuiChartsSurface-root"
          height="160"
          viewBox="0 0 160 160"
          width="160"
        >
          <title />
          <desc />
          <path
            class="MuiGauge-referenceArc css-p1sr87-MuiGauge-referenceArc"
            d="M0,-70A70,70,0,1,1,0,70A70,70,0,1,1,0,-70M0,-56A56,56,0,1,0,0,56A56,56,0,1,0,0,-56Z"
            transform="translate(80, 80)"
          />
          <path
            class="MuiGauge-valueArc css-b9rdri-MuiGauge-referenceArc"
            d="M0,-62.61A7,7,0,0,1,7.778,-69.567A70,70,0,1,1,-34.598,60.852A7,7,0,0,1,-36.801,50.652L-36.801,50.652A7,7,0,0,1,-27.678,48.682A56,56,0,1,0,6.222,-55.653A7,7,0,0,1,0,-62.61Z"
            transform="translate(80, 80)"
          />
          <g
            class="MuiGauge-valueText"
          >
            <text
              dominant-baseline="central"
              text-anchor="middle"
              x="80"
              y="80"
            >
              <tspan
                dominant-baseline="central"
                dy="0px"
                x="80"
              >
                60K
              </tspan>
            </text>
          </g>
        </svg>
      </div>
    </div>
    <div
      class="MuiBox-root css-j7qwjs"
    >
      <div
        class="MuiBox-root css-18zczcl"
      >
        <div
          class="MuiTypography-root MuiTypography-body2 css-1naii7p-MuiTypography-root"
        >
          <div
            class="MuiBox-root css-n9ixjc"
          />
          Social Media
        </div>
        <div
          class="MuiTypography-root MuiTypography-body2 css-1yr8029-MuiTypography-root"
        >
           78%
        </div>
      </div>
      <hr
        class="MuiDivider-root MuiDivider-fullWidth css-1cx35xd-MuiDivider-root"
      />
      <div
        class="MuiBox-root css-18zczcl"
      >
        <div
          class="MuiTypography-root MuiTypography-body2 css-1naii7p-MuiTypography-root"
        >
          <div
            class="MuiBox-root css-1cmcpbx"
          />
          Organic Search
        </div>
        <div
          class="MuiTypography-root MuiTypography-body2 css-1yr8029-MuiTypography-root"
        >
           22%
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
